eo-phi-normalizer/src/Language/EO/Phi/Dataize.hs

```diff
@@ -270,87 +278,129 @@ evaluateBytesBytesFunChain name = evaluateUnaryDataizationFunChain name intToByt
--- evaluateBuiltinFunChain name@"Lorg_eolang_malloc_memory_block_pointer_write" obj = _ -- TODO
--- evaluateBuiltinFunChain name@"Lorg_eolang_malloc_memory_block_pointer_free" obj = _ -- TODO
--- cage
--- evaluateBuiltinFunChain name@"Lorg_eolang_cage_φ" obj = _ -- TODO
--- evaluateBuiltinFunChain name@"Lorg_eolang_cage_encaged_φ" obj = _ -- TODO
--- evaluateBuiltinFunChain name@"Lorg_eolang_cage_encaged_encage" obj = _ -- TODO
--- I/O
--- evaluateBuiltinFunChain name@"Lorg_eolang_io_stdin_next_line" obj = evaluateIODataizationFunChain getLine obj
--- evaluateBuiltinFunChain name@"Lorg_eolang_io_stdin_φ" obj = evaluateIODataizationFunChain getContents obj
--- evaluateBuiltinFunChain name@"Lorg_eolang_io_stdout" obj = evaluateUnaryDataizationFunChain boolToBytes bytesToString wrapBytesInBytes (extractLabel "text") ((`seq` True) . unsafePerformIO . putStrLn) obj
--- others
-evaluateBuiltinFunChain name@"Lorg_eolang_dataized" obj =
-  evaluateUnaryDataizationFunChain name id id wrapBytesInBytes (extractLabel "target") id obj
-evaluateBuiltinFunChain name@"Lorg_eolang_error" obj = evaluateUnaryDataizationFunChain name stringToBytes bytesToString wrapBytesInBytes (extractLabel "message") error obj
--- evaluateBuiltinFunChain name@"Lorg_eolang_seq" obj = _ -- TODO
--- evaluateBuiltinFunChain name@"Lorg_eolang_as_phi" obj = _ -- TODO
--- evaluateBuiltinFunChain name@"Lorg_eolang_rust" obj = _ -- TODO
--- evaluateBuiltinFunChain name@"Lorg_eolang_try" obj = _ -- TODO
-evaluateBuiltinFunChain "Package" obj@(Formation bindings) = do
-  \state -> do
-    fmap dataizePackage getContext >>= \case
-      True -> do
-        let (packageBindings, restBindings) = span isPackage bindings
-        bs <- mapM dataizeBindingChain restBindings
-        logStep "Dataized 'Package' siblings" (Left $ Formation (bs ++ packageBindings))
-        return (Formation (bs ++ packageBindings), state)
-      False ->
-        return (Formation bindings, state)
- where
-  isPackage (LambdaBinding (Function "Package")) = True
-  isPackage _ = False
-  dataizeBindingChain (AlphaBinding attr o) = do
-    ctx <- getContext
-    let extendedContext = (extendContextWith obj ctx){currentAttr = attr}
-    dataizationResult <- incLogLevel $ withContext extendedContext $ dataizeRecursivelyChain False o
-    return (AlphaBinding attr (either id (Formation . singleton . DeltaBinding) dataizationResult))
-  dataizeBindingChain b = return b
-evaluateBuiltinFunChain atomName obj = \state -> do
-  logStep ("[WARNING]: unknown atom (" <> atomName <> ")") (Left obj)
+evaluateBuiltinFunChain name obj =
+  case HashMap.lookup name knownAtomsSet of
+    Just f -> f name obj
+    Nothing -> evaluateBuiltinFunChainUnknown name obj
+
+evaluateBuiltinFunChainUnknown :: String -> a -> b1 -> Chain (Either a b2) (a, b1)
+evaluateBuiltinFunChainUnknown atomName obj state = do
+  logStep [fmt|[WARNING]: unknown atom ({atomName})|] (Left obj)
```

Comment:
Unknown atoms are normal and should not be warnings.

Reply:
Addressed in https://github.com/objectionary/normalizer/pull/426/commits/75fb7691c7e8842a7ae673d77dcccd8142d9bb28